### PR TITLE
migrate code from googleapis/java-dlp

### PR DESCRIPTION
**Pull request by Sita04**
Migrating samples from java-dlp.
Was not part of the mono-repo migration efforts as the java-dlp repository was part GAPIC and part Veneer.

### CODE REVIEW

File: dlp/snippets/install-without-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>dlp-install-without-bom</artifactId>
+  <packaging>jar</packaging>
+  <name>Google Cloud Data Loss Prevention Install Without Bom</name>
+  <url>https://github.com/googleapis/java-dlp</url>
```
Comment (Shabirmean) [inline dlp/snippets/install-without-bom/pom.xml:8]: I think we need to change these?

Comment (Shabirmean) [reply dlp/snippets/install-without-bom/pom.xml:8]: See example here: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/dataproc/pom.xml#L4-L8

File: dlp/snippets/pom.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-dlp-samples</artifactId>
+  <version>0.0.1-SNAPSHOT</version><!-- This artifact should not be released -->
+  <packaging>pom</packaging>
+  <name>Google Cloud Data Loss Prevention Samples Parent</name>
+  <url>https://github.com/googleapis/java-dlp</url>
+  <description>
+    Java idiomatic client for Google Cloud Platform services.
+  </description>
```
Comment (Shabirmean) [inline dlp/snippets/pom.xml:12]: Update here as well like [this](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/dataproc/pom.xml#L4-L8)

File: dlp/snippets/snapshot/pom.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>dlp-snapshot</artifactId>
+  <packaging>jar</packaging>
+  <name>Google Cloud Data Loss Prevention Snapshot Samples</name>
+  <url>https://github.com/googleapis/java-dlp</url>
```
Comment (Shabirmean) [inline dlp/snippets/snapshot/pom.xml:8]: Update details here [as seen here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/dataproc/pom.xml#L4-L8)

File: dlp/snippets/install-without-bom/pom.xml (hunk 1 above)
Comment (Sita04) [reply dlp/snippets/install-without-bom/pom.xml]: Thanks for the catch. Done.

File: dlp/snippets/pom.xml (hunk 2 above)
Comment (Sita04) [reply dlp/snippets/pom.xml:12]: Updated.

File: dlp/snippets/snapshot/pom.xml (hunk 3 above)
Comment (Sita04) [reply dlp/snippets/snapshot/pom.xml]: Updated.

File: dlp/snippets/snippets/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Cloud Data Loss Prevention (DLP) API Samples
+
+<a href="https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/java-dlp&page=editor&open_in_editor=samples/snippets/README.md">
```
Comment (Shabirmean) [inline dlp/snippets/snippets/README.md:3]: Need to update the links here as well!

**Review by Shabirmean — CHANGES_REQUESTED** [review]
- The migration should not bring in the following directories:
    - ***install-without-bom***
    - ***snapshot***

- Everything inside the snippets folder must be top level and the pom outside can be removed

See [FAQ](http://goto.google.com/java-monorepo-migration-faq)

**Comment by Sita04** [comment]
> * The migration should not bring in the following directories:
>   
>   * _**install-without-bom**_
>   * _**snapshot**_
> * Everything inside the snippets folder must be top level and the pom outside can be removed
> 
> See [FAQ](http://goto.google.com/java-monorepo-migration-faq)

Agree that's the norm. But there are some resources inside both those folders. Wasn't sure if those were needed.

**Review by Shabirmean — CHANGES_REQUESTED** [review]
Can you fix the snippet bot errors?

-[dlp/snippets/pom.xml:53](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/8d23d4554ef47e8b8545c1e36137d0dbb158b67b/dlp/snippets/pom.xml#L53), tag dlp_install_with_bom doesn't have a matching start tag.

**Review by Shabirmean — APPROVED** [review]

**Comment by Sita04** [comment]
Couldn't rebase and merge due to >250 commits. Hence, squashing.

**Pull request merged**